lib/view/dayView.dart

```diff
@@ -241,41 +228,36 @@ class TaskCard extends StatefulWidget {
 }
 
 class _TaskCardState extends State<TaskCard> {
-  Color tilecolor = Colors.white;
   TextDecoration dec = TextDecoration.none;
 
   @override
   Widget build(BuildContext context) {
+    Task task = widget.tasksDueToday[widget.index];
+    Color tilecolor = Colors.white;
+    if (task.completed) {
+      tilecolor = Colors.green;
+    }
     return SizedBox(
         height: 40,
         child: Card(
             color: tilecolor,
             child: InkWell(
               splashColor: Colors.blue.withAlpha(30),
-              onTap: () {
-                Navigator.push(
-                  context,
-                  MaterialPageRoute(
-                      builder: (context) => TaskDetailsView(
-                          widget.tasksDueToday[widget.index].name,
-                          widget.tasksDueToday[widget.index].description,
-                          widget.tasksDueToday[widget.index].location)),
-                );
+              onTap: () async {
+                showTaskDetailPopup(context, task);
               },
               onLongPress: () {
                 if (tilecolor == Colors.white) {
-                  widget.tasksDueToday[widget.index].completed =
-                      !widget.tasksDueToday[widget.index].completed;
-                  db.setTask(widget.tasksDueToday[widget.index]);
+                  task.completed = !task.completed;
+                  db.setTask(task);
                   setState(() {
                     tilecolor = Colors.green;
                     dec = TextDecoration.lineThrough;
                   });
                 } else {
                   setState(() {
-                    widget.tasksDueToday[widget.index].completed =
-                        !widget.tasksDueToday[widget.index].completed;
-                    db.setTask(widget.tasksDueToday[widget.index]);
+                    task.completed = !task.completed;
+                    db.setTask(task);
```

Comment:
Would you like backend team to do the database saving for you? we can make the db.setTask call every time you change something instead.

Reply:
I think it's fine as is